screens/UserScreens/UserProfile.jsx

```diff
@@ -31,20 +30,84 @@ export default function UserProfileScreen({ navigation }) {
 
     useEffect(() => {
         get(child(dbRef(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
-            setUserName(snapshot.exists() ? [snapshot.val().firstName, snapshot.val().lastName] : "Anonymous");
+            setUserName(
+                snapshot.exists()
+                    ? [snapshot.val().firstName, snapshot.val().lastName]
+                    : "Anonymous"
+            );
         });
     }, []);
 
     const handleSignOut = () => {
         try {
             signOut(auth);
-        } catch (error) { 
+        } catch (error) {
             alert("sign out failed:", error.message);
         }
     };
 
+    // TODO: Check this works w multiple admins!
     const handleDeleteAccount = () => {
-        alert("Delete Account");
+        try {
+            // Find all of user's productions
+            get(child(dbRef(db), `users/${auth.currentUser.uid}/productions`)).then((snapshot) => {
+                if (snapshot.exists()) {
+                    const productions = snapshot.val();
+                    // Delete user from all productions
+                    Object.keys(productions).forEach((playCode) => {
+                        remove(
+                            dbRef(
+                                db,
+                                `productions/${playCode}/participants/${auth.currentUser.uid}`
+                            )
+                        );
+                        get(child(dbRef(db), `productions/${playCode}/admins`)).then((snapshot) => {
+                            if (snapshot.exists()) {
```

Comment:
if (!snapshot.exists()) {return}
throughout would make this whole block more readable and maintainable